t/rest_server/.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,127 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'deploy'
+
+on:
+  workflow_dispatch:
+    inputs:
+      deploymentEnvironment:
+        type: 'choice'
+        description: 'Environment to deploy image to.'
+        default: 'autopush'
+        options:
+          - 'autopush'
+  workflow_call:
+    inputs:
+      deploymentEnvironment:
+        type: 'string'
+        description: 'Environment to deploy image to.'
+        default: 'autopush'
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+env:
+  # This should be the only env variable defined in this file. Define all other
+  # variables in the bootstrap file.
+  # More environments are expected to be added in the future.
+  AUTOPUSH_ENV_FILE: '.github/config/autopush.env'
+  PRODUCTION_ENV_FILE: '.github/config/production.env'
+
+jobs:
+  image-release:
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    name: 'Release Image'
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: 'Checkout Repo'
+        uses: 'actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab' # ratchet:actions/checkout@v3
+      - name: 'Load variables from Autopush .env file'
+        if: "${{ inputs.deploymentEnvironment == 'autopush' }}"
+        uses: 'abcxyz/pkg/.github/actions/override-workflow-variables@main'
+        with:
+          env_file: '${{ env.AUTOPUSH_ENV_FILE }}'
```

Comment:
Can we simplify away the AUTOPUSH_ENV_FILE variable just by loading `.github/config/${{inputs.deployment_environment}}.env`?

That eliminates the need for:
 - This "if"
 - The `*_ENV_FILE` variables
 - Separate override-workflow-variables steps for each environment

Reply:
Good idea, changed to remove env variable.